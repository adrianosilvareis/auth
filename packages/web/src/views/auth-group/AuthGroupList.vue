<template>
  <AuthGroupTable :items="items" :columns="columns" />
</template>

<script lang="ts">
import { Options, Vue } from "vue-class-component";
import DataTable from "@/components/data-table/DataTable.vue";
import { ColumnsConfig } from "../../components/data-table/protocols/data-table-utils";
import { AuthGroup as AuthGroupModel } from "@auth/entity";

class AuthGroupTable extends DataTable<AuthGroupModel> {}

@Options({
  components: {
    AuthGroupTable,
  },
})
export default class AuthGroupList extends Vue {
  columns: ColumnsConfig<AuthGroupModel> = [
    { title: "Meu Titulo", key: "title" },
    "activities",
  ];

  items: AuthGroupModel[] = [
    {
      id: "1-1-1-1",
      title: "system Admin",
      activities: [
        {
          name: "create user",
          permissions: ["read"],
        },
      ],
    },
  ];
}
</script>
