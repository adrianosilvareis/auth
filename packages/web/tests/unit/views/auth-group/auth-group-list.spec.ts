import { shallowMount, flushPromises } from "@vue/test-utils";
import AuthGroupList from "@/views/auth-group/AuthGroupList.vue";
import { createStore } from "vuex";
import authGroup from "@/store/views/auth-group";

const store = createStore({
  modules: {
    authGroup,
  },
});

describe("AuthGroupList.Vue", () => {
  it("should render auth-group-list on DataTable component", async () => {
    const wrapper = shallowMount(AuthGroupList, {
      global: {
        plugins: [store],
      },
    });
    await flushPromises();
    expect(wrapper.element).toMatchSnapshot();
  });
});
