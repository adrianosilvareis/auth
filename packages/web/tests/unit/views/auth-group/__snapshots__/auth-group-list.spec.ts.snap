// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthGroupList.Vue should render auth-group-list on DataTable component 1`] = `
<auth-group-table-stub
  columns="[object Object],[object Object]"
  items=""
/>
`;
